meson.build

```diff
@@ -0,0 +1,237 @@
+project('libass', 'c', license: 'ISC',
+        meson_version: '>= 0.55.0',
+        # cpp_std specified because meson compiler options are currently global
+        default_options: ['c_std=c99', 'cpp_std=c++11', 'buildtype=debugoptimized', 'warning_level=2'],
+        version: '0.16.0')
+
+conf = configuration_data()
+deps = []
+
+# Compiler setup
+
+cc = meson.get_compiler('c')
+
+if cc.get_argument_syntax() == 'gcc'
+    cflags = [
+        '-Wno-sign-compare',
+        '-Wno-unused-parameter',
+        '-Werror-implicit-function-declaration',
+        '-Wstrict-prototypes',
+        '-Wpointer-arith',
+        '-Wredundant-decls',
+        '-Wno-missing-field-initializers'
+    ]
+    add_project_arguments(cc.get_supported_arguments(cflags), language: 'c')
+endif
+
+test_args = []
+
+if host_machine.system() == 'linux'
+    test_args += '-D_GNU_SOURCE'
+    add_project_arguments('-D_GNU_SOURCE', language: 'c')
+elif host_machine.system() == 'darwin'
+    test_args += '-D_DARWIN_C_SOURCE'
+    add_project_arguments('-D_DARWIN_C_SOURCE', language: 'c')
+elif cc.get_id() != 'msvc'
+    test_args += '-D_POSIX_C_SOURCE=200809L'
+    add_project_arguments('-D_POSIX_C_SOURCE=200809L', language: 'c')
+endif
+
+# Configuration
+
+if get_option('large-tiles')
+    conf.set('CONFIG_LARGE_TILES', 1)
+endif
+
+conf.set('PACKAGE_NAME', 'libass')
+conf.set('PACKAGE_VERSION', meson.project_version())
+
+conf.set('CONFIG_SOURCEVERSION', '"meson, on or after: ' + meson.project_version() + '"')
+
+str_check_functions = [
+    'strdup',
+    'strndup'
+]
+
+foreach name: str_check_functions
+    if cc.has_function(name) and cc.has_header_symbol('string.h', name, args: test_args)
+        conf.set('HAVE_@0@'.format(name.to_upper()), 1)
+    endif
+endforeach
+
+# Dependencies
+
+deps += cc.find_library('m', required: false)
+
+# Replace with Meson iconv lookup (`dependency('iconv')`) once 0.60 can be required, see: https://github.com/libass/libass/pull/330#discussion_r707666730
+iconv_dep = cc.find_library('iconv', required: false)
+if iconv_dep.found()
+    deps += iconv_dep
+endif
+if iconv_dep.found() or cc.has_function('iconv_open')
+    conf.set('CONFIG_ICONV', 1)
+endif
+
+freetype_dep = dependency('freetype2',version: '>= 2.10.92',
```

Comment:
Minimum version is incorrect; should be `>= 9.17.3`.

Reply:
There was a lot of discussion about this before, but I'm fairly sure this version is correct? It's a public vs private version thing.

Reply:
2.10.92 is a project version, 9.17.3 is a pkg-config dependency version.

The current version of freetype2 is 2.12.1, but its pkg-config file specifies: `Version: 24.3.18`.

Reply:
Anyway this should match what the autotools build does ;)
https://github.com/libass/libass/blob/a2b39cde4ecb74d5e6fccab4a5f7d8ad52b2b1a4/configure.ac#L68-L73

Reply:
No, I'm agreeing with you. Meson uses pkg-config for lookup, and subproject wrapped dependencies have a version attribute on the internal dependency which corresponds to the pkg-config version and in freetype's case is different from the version field of the overall project.

Meson ***requires*** to use the library (pkg-config) version for a dependency lookup, this PR is wrong to use the project version.

I was hoping to clarify to @CoffeeFlux why there are two different version numbers.

Reply:
Splendid, thanks for the clarification.

Reply:
Also yeah, 2.10.92 "works" because it version compares as much, much older than any supported version, and allows way too much.

If it were actually looked up as a project version then it would be extremely strict and only allow basically the most recent freetype version published.

Reply:
2.10.92 appears to be simply a copy-paste error: it’s the Fontconfig version. I found this by accident when searching for “2.10” through the GitHub discussion 😅